pkg/backends/cloudwatch/cloudwatch.go

```diff
@@ -157,59 +176,115 @@ func (client Client) buildMetricData(metrics *gostatsd.MetricMap) (metricData []
 	return metricData
 }
 
-// SendMetricsAsync sends the metrics in a MetricsMap to AWS Cloudwatch,
-// preparing payload synchronously but doing the send asynchronously.
-func (client Client) SendMetricsAsync(ctx context.Context, metrics *gostatsd.MetricMap, cb gostatsd.SendCallback) {
-	api := client.cloudwatch
-	metricData := client.buildMetricData(metrics)
-	length := len(metricData)
-	errors := []error{}
+func (client *Client) flush() {
+	client.timerMutex.Lock()
 
-	if length < 1 {
-		cb(errors)
+	// If there already is a timer, we already have a function waiting to execute
+	if client.timer != nil {
+		client.timer.Reset(client.batchDuration)
```

Comment:
The logic on this works because of the mutex, but I did have to work through it, which should be unnecessary cognitive load.

Can you add a comment to the effect that even if `client.timer` has already fired, everything will still process due to the mutex?